### PR TITLE
llcppsymg: refine gosymbol to support method name

### PR DESCRIPTION
**Pull request by tsingbx**

### CODE REVIEW

File: chore/_xtool/llcppsymg/parse/names.go [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+	_, fileName := filepath.Split(incPath)
+	ext := filepath.Ext(fileName)
+	if len(ext) > 0 {
+		fileName = strings.TrimSuffix(fileName, ext)
+	}
+	if strings.HasPrefix(fileName, "_") {
+		fileName = "X" + fileName
+	}
+	return fileName + ".go"
+}
```
Comment (luoliwoshang) [inline chore/_xtool/llcppsymg/parse/names.go]: 这部分文件名的输出算是`gogensig`独有的处理，保留在`gogensig`即可，并不需要复用

File: chore/_xtool/llcppsymg/parse/names.go
Comment (luoliwoshang) [inline chore/_xtool/llcppsymg/parse/names.go:1]: 这里会预期给`gogensig`复用，单独拎在`llcppsymg/names`吧

**Comment by tsingbx** [comment]
> 对于更新的内容，增加对应的测试用例在 `llcppsymg/_cmptest` 下吧

done

**Comment by luoliwoshang** [comment]
> > 对于更新的内容，增加对应的测试用例在 `llcppsymg/_cmptest` 下吧
> 
> done
有个测试没过，这里再check一下吧
```
_xtool/llcppsymg/_cmptest on  remotes/tsingbx/llcppsymg/struct_methodname:llcppsymg/struct_methodname [$?] 
❯ llgo cmptest ./...                                      
# github.com/goplus/llgo/chore/_xtool/llcppsymg/_cmptest/args_test
# github.com/goplus/llgo/chore/_xtool/llcppsymg/_cmptest/clangutils_test
# github.com/goplus/llgo/chore/_xtool/llcppsymg/_cmptest/config_test
# github.com/goplus/llgo/chore/_xtool/llcppsymg/_cmptest/names_test
# github.com/goplus/llgo/chore/_xtool/llcppsymg/_cmptest/parse_test
# github.com/goplus/llgo/chore/_xtool/llcppsymg/_cmptest/symbol_test
# github.com/goplus/llgo/chore/_xtool/llcppsymg/_cmptest/symg_test
=> Result of llgo.expect
#stdout
=== Test Case: inireader ===
[{
                "mangle":       "_ZN9INIReaderC1EPKc",
                "c++":  "INIReader::INIReader(const char *)",
                "go":   "(*Reader).Init"
        }, {
                "mangle":       "_ZN9INIReaderC1EPKcl",
                "c++":  "INIReader::INIReader(const char *, long)",
                "go":   "(*Reader).Init__1"
        }, {
                "mangle":       "_ZN9INIReaderD1Ev",
                "c++":  "INIReader::~INIReader()",
                "go":   "(*Reader).Dispose"
        }, {
                "mangle":       "_ZNK9INIReader10ParseErrorEv",
                "c++":  "INIReader::ParseError()",
                "go":   "(*Reader).ModifyedParseError"
        }, {
                "mangle":       "_ZNK9INIReader3GetEPKcS1_S1_",
                "c++":  "INIReader::Get(const char *, const char *, const char *)",
                "go":   "(*Reader).Get"
        }]
=== Test Case: lua ===
[{
                "mangle":       "lua_error",
                "c++":  "lua_error(lua_State *)",
                "go":   "(*State).Error"
        }, {
                "mangle":       "lua_next",
                "c++":  "lua_next(lua_State *, int)",
                "go":   "(*State).Next"
        }, {
                "mangle":       "lua_concat",
                "c++":  "lua_concat(lua_State *, int)",
                "go":   "(*State).Concat"
        }, {
                "mangle":       "lua_stringtonumber",
                "c++":  "lua_stringtonumber(lua_State *, const char *)",
                "go":   "(*State).Stringtonumber"
        }]

#stderr

#exit 0

=> Expected llgo.expect
#stdout
=== Test Case: inireader ===
[{
                "mangle":       "_ZN9INIReaderC1EPKc",
                "c++":  "INIReader::INIReader(const char *)",
                "go":   "(*Reader).Init"
        }, {
                "mangle":       "_ZN9INIReaderC1EPKcl",
                "c++":  "INIReader::INIReader(const char *, long)",
                "go":   "(*Reader).Init__1"
        }, {
                "mangle":       "_ZN9INIReaderD1Ev",
                "c++":  "INIReader::~INIReader()",
                "go":   "(*Reader).Dispose"
        }, {
                "mangle":       "_ZNK9INIReader10ParseErrorEv",
                "c++":  "INIReader::ParseError()",
                "go":   "(*Reader).ModifyedParseError"
        }, {
                "mangle":       "_ZNK9INIReader3GetEPKcS1_S1_",
                "c++":  "INIReader::Get(const char *, const char *, const char *)",
                "go":   "(*Reader).Get"
        }]
=== Test Case: lua ===
[{
                "mangle":       "lua_error",
                "c++":  "lua_error(lua_State *)",
                "go":   "Error"
        }, {
                "mangle":       "lua_next",
                "c++":  "lua_next(lua_State *, int)",
                "go":   "Next"
        }, {
                "mangle":       "lua_concat",
                "c++":  "lua_concat(lua_State *, int)",
                "go":   "Concat"
        }, {
                "mangle":       "lua_stringtonumber",
                "c++":  "lua_stringtonumber(lua_State *, const char *)",
                "go":   "Stringtonumber"
        }]

#stderr

#exit 0
2024/11/27 15:57:16 checkEqual: unexpected llgo.expect
```

**Comment by tsingbx** [comment]
> > > 对于更新的内容，增加对应的测试用例在 `llcppsymg/_cmptest` 下吧
> > 
> > 
> > done
> 
> 有个测试没过，这里再check一下吧
> 
> ```
> _xtool/llcppsymg/_cmptest on  remotes/tsingbx/llcppsymg/struct_methodname:llcppsymg/struct_methodname [$?] 
> ❯ llgo cmptest ./...                                      
> # github.com/goplus/llgo/chore/_xtool/llcppsymg/_cmptest/args_test
> # github.com/goplus/llgo/chore/_xtool/llcppsymg/_cmptest/clangutils_test
> # github.com/goplus/llgo/chore/_xtool/llcppsymg/_cmptest/config_test
> # github.com/goplus/llgo/chore/_xtool/llcppsymg/_cmptest/names_test
> # github.com/goplus/llgo/chore/_xtool/llcppsymg/_cmptest/parse_test
> # github.com/goplus/llgo/chore/_xtool/llcppsymg/_cmptest/symbol_test
> # github.com/goplus/llgo/chore/_xtool/llcppsymg/_cmptest/symg_test
> => Result of llgo.expect
> #stdout
> === Test Case: inireader ===
> [{
>                 "mangle":       "_ZN9INIReaderC1EPKc",
>                 "c++":  "INIReader::INIReader(const char *)",
>                 "go":   "(*Reader).Init"
>         }, {
>                 "mangle":       "_ZN9INIReaderC1EPKcl",
>                 "c++":  "INIReader::INIReader(const char *, long)",
>                 "go":   "(*Reader).Init__1"
>         }, {
>                 "mangle":       "_ZN9INIReaderD1Ev",
>                 "c++":  "INIReader::~INIReader()",
>                 "go":   "(*Reader).Dispose"
>         }, {
>                 "mangle":       "_ZNK9INIReader10ParseErrorEv",
>                 "c++":  "INIReader::ParseError()",
>                 "go":   "(*Reader).ModifyedParseError"
>         }, {
>                 "mangle":       "_ZNK9INIReader3GetEPKcS1_S1_",
>                 "c++":  "INIReader::Get(const char *, const char *, const char *)",
>                 "go":   "(*Reader).Get"
>         }]
> === Test Case: lua ===
> [{
>                 "mangle":       "lua_error",
>                 "c++":  "lua_error(lua_State *)",
>                 "go":   "(*State).Error"
>         }, {
>                 "mangle":       "lua_next",
>                 "c++":  "lua_next(lua_State *, int)",
>                 "go":   "(*State).Next"
>         }, {
>                 "mangle":       "lua_concat",
>                 "c++":  "lua_concat(lua_State *, int)",
>                 "go":   "(*State).Concat"
>         }, {
>                 "mangle":       "lua_stringtonumber",
>                 "c++":  "lua_stringtonumber(lua_State *, const char *)",
>                 "go":   "(*State).Stringtonumber"
>         }]
> 
> #stderr
> 
> #exit 0
> 
> => Expected llgo.expect
> #stdout
> === Test Case: inireader ===
> [{
>                 "mangle":       "_ZN9INIReaderC1EPKc",
>                 "c++":  "INIReader::INIReader(const char *)",
>                 "go":   "(*Reader).Init"
>         }, {
>                 "mangle":       "_ZN9INIReaderC1EPKcl",
>                 "c++":  "INIReader::INIReader(const char *, long)",
>                 "go":   "(*Reader).Init__1"
>         }, {
>                 "mangle":       "_ZN9INIReaderD1Ev",
>                 "c++":  "INIReader::~INIReader()",
>                 "go":   "(*Reader).Dispose"
>         }, {
>                 "mangle":       "_ZNK9INIReader10ParseErrorEv",
>                 "c++":  "INIReader::ParseError()",
>                 "go":   "(*Reader).ModifyedParseError"
>         }, {
>                 "mangle":       "_ZNK9INIReader3GetEPKcS1_S1_",
>                 "c++":  "INIReader::Get(const char *, const char *, const char *)",
>                 "go":   "(*Reader).Get"
>         }]
> === Test Case: lua ===
> [{
>                 "mangle":       "lua_error",
>                 "c++":  "lua_error(lua_State *)",
>                 "go":   "Error"
>         }, {
>                 "mangle":       "lua_next",
>                 "c++":  "lua_next(lua_State *, int)",
>                 "go":   "Next"
>         }, {
>                 "mangle":       "lua_concat",
>                 "c++":  "lua_concat(lua_State *, int)",
>                 "go":   "Concat"
>         }, {
>                 "mangle":       "lua_stringtonumber",
>                 "c++":  "lua_stringtonumber(lua_State *, const char *)",
>                 "go":   "Stringtonumber"
>         }]
> 
> #stderr
> 
> #exit 0
> 2024/11/27 15:57:16 checkEqual: unexpected llgo.expect
> ```
done

**Pull request closed without merging**